src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Carnivore) {
+      return;
+    }
+
+    if (target.hidden !== true && target instanceof Herbivore) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
```

Comment:
When health reaches 0 or below, the animal should be removed from the 'Animal.alive' array. Ensure that the health check is '<= 0' to cover cases where health might go below 0 due to some future changes in the bite logic.